bin/add-server

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env node
+
+require('dotenv').config();
+const axios = require('axios');
+
+// step 1, check command line args
+const args = process.argv.slice(2); // first 2 are "node add-server"
+
+const newServer = {};
+
+for (let i = 0 ; i < args.length ; i++) {
+  const argKey = args[i];
+  const argValue = args[++i];
+
+  switch (argKey) {
+    case '-e':
+    case '--endpoint':
+      newServer['endpoint'] = argValue;
+      break;
+
+    case '-t':
+    case '--type':
+      newServer['type'] = argValue;
+      break;
+
+    case '-c':
+    case '--clientId':
+      newServer['clientId'] = argValue;
+      break;
+
+    case '-n':
+    case '--name':
+      newServer['name'] = argValue;
+      break;
+
+    default:
+      console.error(`Unknown command line option: ${argKey}`);
+      console.error(`Full args: ${args}`);
+      process.exit(1);
+  }
+}
+
+if (!newServer['endpoint']) {
+  console.error("Required argument -e|--endpoint missing!");
+  process.exit(1);
+}
+
+if (!newServer['name']) {
+  // if a name wasn't provided, use the endpoint URL as the name
+  newServer['name'] = newServer['endpoint'];
+}
+
+const baseURL = process.env.BASE_URL;
+// assumption is that BASE_URL does not contain a trailing /
+
+const serversURL = baseURL + '/servers';
+
+// see if the server is already running
+// by hitting the base url.
+// no connection = it's not running, add to DB directly
+// connection succeeds = it's running, use the API
+axios.get(serversURL, {
+  headers: {
+    'Authorization': 'Bearer admin'
+  }
+}).then(callApi).catch(addToDb);
+
+function addToDb() {
+  console.log('App not running. Adding server to db directly.');
+
+  const { runWhenDBReady } = require('../src/storage/postinit');
+  const servers = require('../src/storage/servers');
+
+  runWhenDBReady(() => {
+    servers.addServer(newServer);
```

Comment:
When adding the server this way we loose out on the `id` field.... perhaps its best to have the `addServer` method check for an `id` and add one if it doesn't already exist on the server.

Reply:
I'll move the "add ID if not present" logic from the /servers route to the addServer method. This will also fix the issue with it seemingly not working when you run it twice